python/ray/_private/async_compat.py

```diff
@@ -29,3 +43,118 @@ async def wrapper(*args, **kwargs):
         return func(*args, **kwargs)
 
     return wrapper
+
+
+try:
+    from contextlib import asynccontextmanager
+except ImportError:
+    # Copy from https://github.com/python-trio/async_generator
+    # for compatible with Python 3.6
```

Comment:
@rkooo567 The Python 3.6 does not support `asynccontextmanager`. This code is copied from `async_generator`.